sdk/state/payment.go

```diff
@@ -110,16 +110,16 @@ var ErrUnderfunded = fmt.Errorf("account is underfunded to make payment")
 // there are additional verifications ConfirmPayment performs that are based
 // on the state of the close agreement signatures.
 func (c *Channel) validatePayment(ca CloseAgreement) (err error) {
-	// If a coordinated close has been accepted already, error.
-	if !c.latestAuthorizedCloseAgreement.isEmpty() && c.latestAuthorizedCloseAgreement.Details.ObservationPeriodTime == 0 &&
-		c.latestAuthorizedCloseAgreement.Details.ObservationPeriodLedgerGap == 0 {
-		return fmt.Errorf("cannot confirm payment after an accepted coordinated close")
-	}
 	// If a coordinated close has been proposed by this channel already, error.
 	if !c.latestUnauthorizedCloseAgreement.isEmpty() && c.latestUnauthorizedCloseAgreement.Details.ObservationPeriodTime == 0 &&
 		c.latestUnauthorizedCloseAgreement.Details.ObservationPeriodLedgerGap == 0 {
 		return fmt.Errorf("cannot propose payment after proposing a coordinated close")
 	}
+	// If a coordinated close has been accepted already, error.
+	if !c.latestAuthorizedCloseAgreement.isEmpty() && c.latestAuthorizedCloseAgreement.Details.ObservationPeriodTime == 0 &&
+		c.latestAuthorizedCloseAgreement.Details.ObservationPeriodLedgerGap == 0 {
+		return fmt.Errorf("cannot confirm payment after an accepted coordinated close")
+	}
```

Comment:
Why did these need to be reordered? The commit is named "fix test", but I can't see how reordering these would result in changed behavior.

Reply:
technically it was already impossible to propose a new payment if you proposed a coordinated close and it's pending
1. if you propose a payment with `O` period of not zero to match the `latestAuthorized` details, you'll fail the compare against `latestUnauthorized` check
2. if you propose a payment with `O` period of zero to match the new `latestUnauthorized` details, you'll fail the compare against `latestAuthorized` check

so to make the check explicit it has to go first 😄